### PR TITLE
Include database-scoped credentials when deploying with default options

### LINKED ISSUE #12839: Database-Scoped Credentials are excluded from project deployment
A recent change uses SchemaCompare default options for deployment, which excludes several object types (including database-scoped credentials).  Deploying a dacpac via the DacFx wizard still works.

### PR DESCRIPTION
**Pull request by Benjin**
Fixes #12839

### CODE REVIEW

**Review by kisantia — APPROVED** [review]
can you also create an issue to track the longer-term fix of adding a new API for the non-schema compare default DacFx deploy options and also checking if the SSDT default deploy options are different?

File: extensions/sql-database-projects/src/dialogs/publishDatabaseDialog.ts [hunk 1]
```diff
@@ -211,10 +211,14 @@ export class PublishDatabaseDialog {
...
 	private async getDeploymentOptions(): Promise<DeploymentOptions> {
 		// eventually, database options will be configurable in this dialog
-		// but for now,  just send the default DacFx deployment options if no options were loaded from a publish profile
+		// but for now, just send the default DacFx deployment options if no options were loaded from a publish profile
 		if (!this.deploymentOptions) {
 			this.deploymentOptions = await utils.GetDefaultDeploymentOptions();
 
+			// re-include database-scoped credentials
+			// 53 comes from $\DacFx_Preview_Dev\Product\Source\DeploymentApi\ObjectTypes.cs (exact value from decompiled metadata)
+			this.deploymentOptions.excludeObjectTypes = this.deploymentOptions.excludeObjectTypes.filter(x => x !== 53);
```
Comment (udeeshagautam) [inline extensions/sql-database-projects/src/dialogs/publishDatabaseDialog.ts:220]: why not SchemaObjectType.DatabaseScopedCredentials we have the value defined in mssql.d.ts

Comment (Benjin) [reply extensions/sql-database-projects/src/dialogs/publishDatabaseDialog.ts:220]: Shoot; clicked too quickly.  Will send again.

**Comment by Benjin** [comment]
> can you also create an issue to track the longer-term fix of adding a new API for the non-schema compare default DacFx deploy options and also checking if the SSDT default deploy options are different?

Created https://github.com/microsoft/azuredatastudio/issues/12847

**Pull request merged**